components/brave_wallet/browser/zcash/zcash_blocks_batch_scan_task.cc

```diff
@@ -130,10 +132,12 @@ void ZCashBlocksBatchScanTask::OnGetFrontierBlock(
 }
 
 void ZCashBlocksBatchScanTask::DownloadBlocks() {
-  uint32_t start_index =
-      downloaded_blocks_ ? from_ + downloaded_blocks_->size() : from_;
-  uint32_t end_index = std::min(start_index + kBlockDownloadBatchSize - 1, to_);
-  auto expected_size = end_index - start_index + 1;
+  uint32_t downloaded_blocks_count =
+      downloaded_blocks_ ? downloaded_blocks_->size() : 0;
+  uint32_t remaining_blocks = scan_range_.count - downloaded_blocks_count;
+  uint32_t expected_size = std::min(kBlockDownloadBatchSize, remaining_blocks);
+  uint32_t start_index = scan_range_.from + downloaded_blocks_count;
+  uint32_t end_index = start_index + expected_size - 1;
 
   context_->zcash_rpc->GetCompactBlocks(
       context_->chain_id, start_index, end_index,
```

Comment:
out of scope by now, but GetCompactBlocks also should follow (start, count) pattern

Reply:
We still need to convert somewhere to {from, to} format since this what RPC API takes